src/editors/containers/VideoUploadEditor/VideoUploader.jsx

```diff
@@ -4,33 +4,62 @@ import { useIntl } from '@edx/frontend-platform/i18n';
 import * as hooks from './hooks';
+import * as editorHooks from '../EditorContainer/hooks';
 import messages from './messages';
 
 const URLUploader = () => {
+  const [textInputValue, setTextInputValue] = React.useState('');
+  const onURLUpload = hooks.onVideoUpload();
   const intl = useIntl();
   return (
-    <div className="d-flex flex-column flex-wrap">
-      <div className="justify-content-center align-self-center bg-light rounded-circle p-4">
-        <Icon src={FileUpload} className="text-muted" />
+    <div className="d-flex flex-column">
+      <div style={{ backgroundColor: '#F2F0EF' }} className="justify-content-center align-self-center rounded-circle p-5">
+        <Icon src={FileUpload} className="text-muted" size="lg" />
       </div>
       <div className="d-flex align-self-center justify-content-center flex-wrap flex-column pt-5">
-        <span style={{ fontSize: '1.35rem' }}>{intl.formatMessage(messages.dropVideoFileHere)}</span>
+        <span style={{ fontSize: '0.875rem' }}>{intl.formatMessage(messages.dropVideoFileHere)}</span>
         <span className="align-self-center" style={{ fontSize: '0.8rem' }}>{intl.formatMessage(messages.info)}</span>
       </div>
-      <div className="align-self-center justify-content-center mx-2 text-dark">OR</div>
+      <div style={{ fontSize: '0.75rem' }} className="align-self-center justify-content-center mx-2 text-dark font-weight-normal">OR</div>
+      <div className="zindex-9 video-id-prompt p-4">
+        <InputGroup>
+          <FormControl
+            placeholder={intl.formatMessage(messages.pasteURL)}
+            aria-label={intl.formatMessage(messages.pasteURL)}
+            aria-describedby="basic-addon2"
+            borderless
+            onClick={(event) => { event.stopPropagation(); }}
+            onChange={(event) => { setTextInputValue(event.target.value); }}
+          />
+          <div style={{ backgroundColor: '#F2F0EF' }} className="justify-content-center align-self-center bg-light rounded-circle p-0 x-small url-submit-button">
+            <IconButton
+              className="text-muted"
+              alt={intl.formatMessage(messages.submitButtonAltText)}
+              src={ArrowForward}
+              iconAs={Icon}
+              size="inline"
+              onClick={(event) => {
+                event.stopPropagation();
+                if (textInputValue.trim() !== '') {
+                  onURLUpload(textInputValue);
+                }
+              }}
+            />
+          </div>
+        </InputGroup>
+      </div>
     </div>
   );
 };
 
-export const VideoUploader = ({ setLoading }) => {
-  const [textInputValue, setTextInputValue] = React.useState('');
-  const onURLUpload = hooks.onVideoUpload();
-  const intl = useIntl();
+export const VideoUploader = ({ setLoading, onClose }) => {
   const dispatch = useDispatch();
+  const intl = useIntl();
+  const handleCancel = editorHooks.handleCancel({ onClose });
```

Comment:
Sorry to bring up a previous nit, but I really don't understand the point of this hook. 

The `handleCancel` function starts with:
```JavaScript
if (onClose) {
    return onClose;
  }
```
So if `onClose` has a value, it will just return that. In the above code `onClose` will always have a value since `onClose` is a required attribute, so the rest of the function will never trigger. 

So this is effectively equivalent to:
```JavaScript
  const handleCancel = onClose;
```

If `onClose` is indeed optional here, then it should be marked as optional on `VideoUploader`, and `VideoUploadEditor`.

Reply:
@xitij2000 I think it's more about passing the function and it is called from editor hook.